src/Symfony/Component/EventDispatcher/DependencyInjection/RegisterListenersPass.php

```diff
@@ -22,12 +22,30 @@
 
 /**
  * Compiler pass to register tagged services for an event dispatcher.
+ *
+ * @psalm-type BeforeAfterDefinition = string|array{0: string, 1: string}
+ * @psalm-type ListenerDefinition = array{serviceId: string, event: string, method: string, before?: BeforeAfterDefinition, after?: BeforeAfterDefinition, priority?: int, dispatchers: list<string>}
+ * @psalm-type AllListenersDefinition = array<string, list<ListenerDefinition>>
  */
 class RegisterListenersPass implements CompilerPassInterface
 {
     private array $hotPathEvents = [];
     private array $noPreloadEvents = [];
 
+    /**
+     * $allListenersMap['event_name']['listener_service_id']['method'] => $listenerDefinition.
+     *
+     * @var array<string, array<string, array<string, ListenerDefinition>>>
+     */
+    private array $allListenersMap = [];
+
+    /**
+     * $listenerClassesMap['listener_FQCN']['event_name'][] => array{serviceId: string, method:string}.
+     *
+     * @var array<string, array<string, list<array{serviceId: string, method:string}>>>
+     */
+    private array $listenerClassesMap = [];
```

Comment:
state stored during the processing **must** be released at the end of the processing.

Or even better, avoid storing state in properties but pass those maps as argument to the methods needing them.

Reply:
done, I've removed the state, but now have much more parameters (`computeBeforeAfterPriorities()` have 5 params)